acestep/core/generation/device_mapping.py

```
@@ -0,0 +1,78 @@
+def _rank_cuda_devices_by_free_vram() -> list[int]:
+    """Return CUDA device indices sorted by descending free VRAM."""
+    if torch is None or not torch.cuda.is_available():
+        return []
+
+    device_count = torch.cuda.device_count()
+    free_by_device: list[tuple[int, int]] = []
+    for idx in range(device_count):
+        free_bytes = 0
+        try:
+            free_bytes = int(torch.cuda.mem_get_info(idx)[0])
+        except Exception:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)